modules/core/shared/src/main/scala/gem/GuideGroup.scala

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+import cats._
+import cats.implicits._
+import cats.data.NonEmptyList
+import gem.enum.Guider
+
+/**
+ * Associates one or more guide probes or guide windows with guide stars.
+ */
+sealed trait GuideGroup extends Product with Serializable {
+
+  def guideStars: NonEmptyList[(Guider, Target)]
+
+}
+
+object GuideGroup {
+
+  /** Group identifier. */
+  final case class Id(toInt: Int) extends AnyVal
+
+  object Id {
+    /** Ids ordered by wrapped integer value. */
+    implicit val IdOrder: Order[Id] =
+      Order.by(_.toInt)
+  }
+
+  /**
+   * Guide "group" supertype for the majority of cases where there is a single
+   * guide star.
+   */
+  sealed trait SingleGuider extends GuideGroup {
+
+    def guider: Guider
+    def target: Target
+
+    override val guideStars: NonEmptyList[(Guider, Target)] =
+      NonEmptyList.one(guider -> target)
+
+  }
+
+  sealed trait Flamingos2 extends SingleGuider
```

Comment:
You might try with `sealed abstract class Flamingos2(val guider: Guider) ...` and `final case class OI(target: Target) extends Flamingos2(Guider.F2OI)` and see if that looks any better.